torch/_dynamo/output_graph.py

```diff
@@ -1020,6 +1021,12 @@ def compile_and_call_fx_graph(self, tx, rv, root):
             "%s", LazyString(lambda: self.get_graph_sizes_log_str(name))
         )
         self.call_cleanup_hooks()
+        if not self.export:
+            prior_fake_mode = self.tracing_context.fake_mode
+            self.tracing_context.fake_mode = FakeTensorMode(
+                shape_env=prior_fake_mode.shape_env,
+                policy_cache=prior_fake_mode.policy_cache,
+            )
```

Comment:
This needs to be scoped to restore on exit

Reply:
(I'm not sure I actually want this logic here at all, but in general, a bare `some_global_state = blah` should set off warning bells in your head)